app/models/debt_balance.rb

```diff
@@ -100,7 +100,18 @@ def self.search(search)
   private
 
   def start_pay_date
+    if debt_id.nil?
+      errors.add(:debt, 'must not be emtpy')
+      return
+    end
+
+    if due_date.nil?
+      errors.add(:due_date, 'must not be empty')
```

Comment:
The easiest way, is just to add a check in the validation itself. Since you already know that those value are tested, it is safe to put something like
```
def start_pay_date
  unless debt && due_date
    return
  do your validations
end
```

or you can do something like

`validate :start_pay_date, if: proc { |k| k.debt && k.due_date }` see [example in IncomeSource](https://github.com/iDreamOn/mymoney/blob/master/app/models/income_source.rb#L10)